src/Sentry/Internal/UnsampledTransaction.cs

```diff
@@ -0,0 +1,103 @@
+using Sentry.Extensibility;
+
+namespace Sentry.Internal;
+
+/// <summary>
+/// We know already, when starting a transaction, whether it's going to be sampled or not. When it's not sampled, we can
+/// avoid lots of unecessary processing. The only thing we need to track is the number of spans that would have been
+/// created (the client reports detailing discarded events includes this detail).
+/// </summary>
+internal sealed class UnsampledTransaction : NoOpTransaction
+{
+    // Although it's a little bit wasteful to create separate individual class instances here when all we're going to
+    // report to sentry is the span count (in the client report), SDK users may refer to things like
+    // `ITransaction.Spans.Count`, so we create an actual collection
```

Comment:
what's the use case for users to get count of spans from transactions?

the goal is to deprecate transactions in the long term. Spans are expected to stream out of the SDK
Anything done absed on span count is going to break so I wouldn't optimzie for that

Reply:
It's for the client report. At the moment the sampling is done at the transaction level... so if a transaction gets sampled out we're sampling out the transaction + it's tree of child spans. We report the number of spans we sampled out in the client report.